### PR TITLE
Test names again

### PR DESCRIPTION
**Pull request by jtlap**
adding values test to simd and correcting some limit bugs

### CODE REVIEW

File: test/function/scalar/fms.cpp [hunk 1]
```diff
@@ -4,7 +4,7 @@
   Copyright 2016 NumScale SAS
 
   Distributed under the Boost Software License, Version 1.0.
-  (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
+  (See accompanying file LICENSE.md or copy fat http://boost.org/LICENSE_1_0.txt)
```
Comment (jfalcou) [inline test/function/scalar/fms.cpp]: Check your commti instead of committing noise please.

File: include/boost/simd/arch/common/simd/function/ffs.hpp [hunk 2]
```diff
@@ -83,7 +83,7 @@ namespace boost { namespace simd { namespace ext
     using result = bd::as_integer_t<A0>;
     BOOST_FORCEINLINE result operator()( const A0& a0) const BOOST_NOEXCEPT
     {
-      result v = firstbitset(a0);
+      result v =  bitwise_cast<result>(firstbitset(a0));
```
Comment (jfalcou) [inline include/boost/simd/arch/common/simd/function/ffs.hpp:86]: Why is this necessary in the first place ? Shold'nt you return the proper type as result ?

File: test/function/simd/ffs.cpp [hunk 3]
```diff
@@ -53,3 +61,63 @@ STF_CASE_TPL("Check ffs on pack" ,  STF_NUMERIC_TYPES)
...
+
+  using bs::ffs;
+  using p_t = bs::pack<T>;
+  using r_t = decltype(ffs(p_t()));
+
+  // return type conformity test
+  STF_TYPE_IS(r_t, (bd::as_integer_t<p_t>));
+
+  // specific values tests
+  STF_EQUAL(ffs(bs::One<p_t>()), bs::One<r_t>());
```
Comment (jfalcou) [inline test/function/simd/ffs.cpp:120]: i think testing ffs and its ilk on more than 0 and 1 to be useful , I think like testing Something like Signmask or somethign non trivial.

File: test/function/simd/divides.regular.cpp [hunk 4]
```diff
@@ -11,6 +11,12 @@
 #include <boost/simd/function/saturated.hpp>
 #include <boost/simd/meta/cardinal_of.hpp>
 #include <simd_test.hpp>
+#include <boost/simd/function/fast.hpp>
+#include <boost/simd/function/saturated.hpp>
```
Comment (jfalcou) [inline test/function/simd/divides.regular.cpp:15]: This is div.regualr, why are you STILL testing fast and saturated here.

File: test/function/simd/acosd.regular.cpp [hunk 5]
```diff
@@ -41,14 +52,14 @@ STF_CASE_TPL("Check acosd on pack" , STF_IEEE_TYPES)
...
+STF_CASE_TPL (" acosd",  STF_IEEE_TYPES)
 {
   namespace bs = boost::simd;
   namespace bd = boost::dispatch;
-
+  using bs::acosd;
   using p_t = bs::pack<T>;
 
-  using r_t = decltype(bs::acosd(p_t()));
+  using r_t = decltype(bs::accurate_(acosd)(p_t()));
```
Comment (jfalcou) [inline test/function/simd/acosd.regular.cpp:59]: we're in .regular, why are you testing accurate ?

File: test/function/simd/acosh.cpp [hunk 6]
```diff
@@ -9,6 +9,11 @@
 #include <simd_test.hpp>
 #include <boost/simd/function/acosh.hpp>
 #include <boost/simd/pack.hpp>
+#include <boost/simd/constant/inf.hpp>
+#include <boost/simd/constant/minf.hpp>
+#include <boost/simd/constant/nan.hpp>
```
Comment (jfalcou) [inline test/function/simd/acosh.cpp:14]: Where they needed ? Also for 0,1,2 etc,, p_t(0) is enough, no need to pull the constant.

File: include/boost/simd/function/definition/dot.hpp [hunk 7]
```diff
@@ -20,7 +20,7 @@ namespace boost { namespace simd
 {
   namespace tag
   {
-    BOOST_DISPATCH_MAKE_TAG(ext, dot_, boost::dispatch::elementwise_<dot_>);
+    BOOST_DISPATCH_MAKE_TAG(ext, dot_, boost::dispatch::abstract_<dot_>);
```
Comment (jfalcou) [inline include/boost/simd/function/definition/dot.hpp:23]: Where is the common implementation we spoke of ?

**Review by jfalcou — COMMENTED** [review]
Three points  :  
- read this carefully : http://chris.beams.io/posts/git-commit/ Your commit messages are but useful.
- It doesn't take more than 5s to structure your commit _per function_ instead of globbing everything alphabetically, missing half of them and then making useless noisy commit

This is supposed to clean things up btu it's so messy I can't be sure it does not in fact rasie the mess level. I'm scheduling this for 4.16.11 cause I don't want silent regression to go in 4.16.10

I would really like you take time and care to submit your PRs so I don't have to spend as much time as reviewing them.

File: test/function/simd/divides.regular.cpp (hunk 4 above)
Comment (jtlap) [reply test/function/simd/divides.regular.cpp:15]: As I have said to you (and I think you agreed) I have added value tests but not removed any existing part in the simd files. They are to be wiped.

File: include/boost/simd/arch/common/simd/function/ffs.hpp (hunk 2 above)
Comment (jtlap) [reply include/boost/simd/arch/common/simd/function/ffs.hpp:86]: Up to now firstbitset always returns an unsigned integral type. ffs always returns a signed  integral type. However this is a choice that can be changed...

File: test/function/simd/ffs.cpp (hunk 3 above)
Comment (jtlap) [reply test/function/simd/ffs.cpp:120]: It's done !

...
  STF_EQUAL(ffs(bs::Signmask< p_t >()), r_t(sizeof(T)*8));

Comment (jfalcou) [reply test/function/simd/ffs.cpp:120]: OK

File: test/function/simd/divides.regular.cpp (hunk 4 above)
Comment (jfalcou) [reply test/function/simd/divides.regular.cpp:15]: OK

File: include/boost/simd/arch/x86/avx/simd/function/rsqrt.hpp [hunk 8]
```diff
@@ -43,7 +47,11 @@ namespace boost { namespace simd { namespace ext
      BOOST_FORCEINLINE A0 operator()(A0 const& a0) const
       {
         A0 const nr  = _mm256_rsqrt_ps( a0 );
-        return nr * Half<A0>() * fnms(a0, sqr(nr), Three<A0>());
+#ifndef BOOST_SIMD_NO_INFINITIES
+        return if_zero_else(a0 == Inf<A0>(), if_else(is_eqz(a0), Inf<A0>(), nr * Half<A0>() * fnms(a0, sqr(nr), Three<A0>())));
```
Comment (jfalcou) [inline include/boost/simd/arch/x86/avx/simd/function/rsqrt.hpp]: I think the `if_else(is_eqz(a0), Inf<A0>(), nr * Half<A0>() * fnms(a0, sqr(nr), Three<A0>()))` could be factorizeed out in a temp variable

Comment (jtlap) [reply include/boost/simd/arch/x86/avx/simd/function/rsqrt.hpp]: done

File: test/function/simd/toint.regular.cpp [hunk 9]
```diff
@@ -42,3 +49,58 @@ STF_CASE_TPL("Check toint on pack" , STF_IEEE_TYPES)
...
+  namespace bd = boost::dispatch;
+  using bs::toint;
+  using p_t = bs::pack<T>;
+  using r_t = decltype(toint(p_t()));
+
+  // return type conformity test
+  STF_TYPE_IS(r_t, (bd::as_integer_t<p_t, signed>));
+
+  // specific values tests
+  STF_EQUAL(toint(bs::One<p_t>()),  bs::One<r_t>());
```
Comment (jfalcou) [inline test/function/simd/toint.regular.cpp:88]: I think the correct basic stuff to check is is more Half Third and so on.

File: test/function/simd/toint.saturated.cpp [hunk 10]
```diff
@@ -42,3 +49,75 @@ STF_CASE_TPL("Check toint_s on pack" , STF_IEEE_TYPES)
...
+  namespace bs = boost::simd;
+  namespace bd = boost::dispatch;
+
+  using r_t = decltype(bs::saturated_(bs::toint)(T()));
+  // return type conformity test
+  STF_TYPE_IS(r_t, (bd::as_integer_t<T, signed>));
+
+
+  // specific values tests
+  STF_EQUAL(bs::saturated_(bs::toint)(bs::Mone<T>()), bs::Mone<r_t>());
```
Comment (jfalcou) [inline test/function/simd/toint.saturated.cpp]: does this give us any info ? you better check something that actually saturate

File: include/boost/simd/arch/common/simd/function/ffs.hpp (hunk 2 above)
Comment (jfalcou) [reply include/boost/simd/arch/common/simd/function/ffs.hpp:86]: Why is ffs return type signed, do we return -1 for some reason ?
